lucille-core/src/main/java/com/kmwllc/lucille/core/RunnerManager.java

```diff
@@ -1,80 +1,212 @@
+   * Method to create a lucille run with a custom configuration.
    *
-   * @return boolean representing whether the lucille run was initiated or not. This will return false if and only if the lucille
-   * run was skipped due to the previous run still existing.
+   * @param runId the unique ID for the lucille run
+   * @param config the configuration to use for the run
+   * @return true if the run was started successfully; false if a run with the given ID is already
+   *         in progress
    */
-  synchronized public boolean run() {
-    return runWithConfig(ConfigFactory.load());
+  public synchronized String createConfig(Config config) {
+    String configId = UUID.randomUUID().toString();
+    configMap.put(configId, config);
+    return configId;
   }
 
   /**
-   * Internal abstraction used to support testing.
+   * Method to start a lucille run with a custom configuration.
+   *
+   * @param runId the unique ID for the lucille run
+   * @param config the configuration to use for the run
+   * @return RunDetails
+   * @throws RunnerManagerException
    */
-  synchronized protected boolean runWithConfig(Config config) {
-    if (isRunning()) {
-      log.warn("Skipping new run; previous lucille run is still in progress.");
-      return false;
+  public synchronized RunDetails runWithConfig(String runId, String configId)
+      throws RunnerManagerException {
+
+    if (runId == null) {
+      throw new RunnerManagerException("runId cannot be null");
+    }
+    
+    if (runDetailsMap.containsKey(runId)) {
+      throw new RunnerManagerException("Run with runId " + runId + " already defined - cannot use this runId again");
+    }
+
+    final RunDetails runDetails = new RunDetails(runId, configId);
+    runDetailsMap.put(runId, runDetails);
+
+    if (isRunning(runId)) {
+      log.warn("Skipping new run with ID '{}'; previous lucille run is still in progress.", runId);
+      return runDetails;
+    }
+
+    final Config config = configMap.get(configId);
+    if (config == null) {
+      log.error("Config with id {} not found", configId);
+      RunnerManagerException exception = new RunnerManagerException("Config with id " + configId + " not found");
+      runDetails.setError(exception);
+      throw exception;
     }
 
-    future = CompletableFuture.runAsync(() -> {
+    runDetails.setFuture(CompletableFuture.runAsync(() -> {
       try {
-        log.info("Starting lucille run via the Runner Manager.");
-        log.info(config.entrySet().toString());
+        log.info("Starting lucille run with ID '{}' via the Runner Manager using configId {}.",
+            runId, configId);
+
+        // For now, we will always use local mode without Kafka
+        runDetails.setRunType(Runner.getRunType(false, true));
 
-        // For now we will always use local mode without kafka
-        RunType runType = Runner.getRunType(false, true);
 
-        Runner.runWithResultLog(config, runType);
+        log.info(config.entrySet().toString());
```

Comment:
hmm... why are we logging the entire config here at info level?
this is going to translate into a lot of redundant log output if the same config is used multiple times
change to log.debug or remove

Reply:
updated to debug